mlflow/tracking/client.py

```diff
@@ -412,6 +423,131 @@ def create_trace(
             tags=tags,
         )
 
+    def start_trace(
+        self,
+        name: str,
+        attributes: Optional[Dict[str, str]] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> MLflowSpanWrapper:
+        """
+        Create a new trace object and start a root span under it.
+
+        This is an imperative API to manually create a new span under a specific trace id and parent
+        span, unlike the higher-level APIs like `@mlflow.trace()` and `with mlflow.start_span()`,
+        which automatically manage the span lifecycle and parent-child relationship.
+
+        A trace started with this method must be ended by calling `end_trace(trace_id)`.
+
+        Args:
+            name: The name of the span.
+            span_type: The type of the span. Can be either a string or a SpanType enum value.
+            parent: The parent span. If None, the span to be created is a root span.
+
+        Returns:
+            str: The ID of the newly created trace.
+
+        Example:
+
+        .. code-block:: python
+
+                from mlflow.tracking import MlflowClient
+
+                client = MlflowClient()
+
+                root_span = client.start_trace("my_trace")
+                trace_id = root_span.trace_id
+
+                # Create a child span
+                child_span = client.start_span(
+                    "child_span", trace_id=trace_id. parent_span_id=root_span.span_id
+                )
+                child_span.set_attributes({"key": "value"})
+                child_span.end()
+
+                client.end_trace(trace_id)
+        """
+        root_span = self._trace_manager.start_detached_span(name)
+
+        trace_info = self._trace_manager.get_trace_info(root_span.trace_id)
+        trace_info.attributes = attributes or {}
+        trace_info.tags = tags or {}
+
+        return root_span
```

Comment:
note: Returning root span because (1) users need `trace_id` and the `span_id` of the root span for creating children spans (2) users may want to set input/output/attributes to the root span